pyfar/dsp/dsp.py

```diff
@@ -1386,6 +1386,232 @@ def time_shift(signal, shift, unit='samples'):
+        ...     start_samples/ir.sampling_rate*1e3,
+        ...     color='k', linestyle='-.', label='start sample')
+        >>> ax.legend()
+
+    """
+    n = int(np.ceil((N+2)/2))
+
+    start_samples = np.zeros(ir.cshape)
+    for ch in np.ndindex(ir.cshape):
+        # Calculate the correlation between the impulse response and its
+        # minimum phase equivalent. This requires a minimum phase equivalent
+        # in the strict sense, instead of the appriximation implemented in
+        # pyfar.
+        ir_minphase = sgn.minimum_phase(ir.time[ch], n_fft=4*ir.n_samples)
+        correlation = sgn.correlate(
+            ir.time[ch],
+            np.pad(ir_minphase, (0, ir.n_samples - (ir.n_samples + 1)//2)),
+            mode='full')
+        lags = np.arange(-ir.n_samples + 1, ir.n_samples)
+
+        # calculate the analytic signal of the correlation function
+        correlation_analytic = sgn.hilbert(correlation)
+
+        # find the maximum of the analytic part of the correlation function
+        # and define the search range around the maximum
+        argmax = np.argmax(np.abs(correlation_analytic))
+        search_region_range = np.arange(argmax-n, argmax+n)
+        search_region = np.imag(correlation_analytic[search_region_range])
+
+        # mask values with a negative gradient
+        mask = np.gradient(search_region, search_region_range) > 0
+
+        # fit a polygon and estimate its roots
+        search_region_poly = np.polyfit(
+            search_region_range[mask]-argmax, search_region[mask], N)
+        roots = np.roots(search_region_poly)
+
+        # Use only real-valued roots
+        if np.all(np.isreal(roots)):
+            root = roots[np.abs(roots) == np.min(np.abs(roots))]
+            start_sample = lags[argmax] + root
+        else:
+            start_sample = np.nan
+            warnings.warn(f"Starting sample not found for channel {ch}")
+
+        start_samples[ch] = start_sample
+
+    return start_samples
+
+
+def find_impulse_response_start(
+        impulse_response,
+        threshold=20,
+        noise_energy='auto'):
+    """Find the first sample of an impulse response comliant to ISO 3382 [#]_.
+    The start sample is identified as the first sample which has a level of
+    20 dB below the maximum of the impulse response. The peak to noise ratio
+    has to be larger or equal this threshold. The function further tries to
+    consider oscillations before the time below the threshold value. This
+    function is primary intended to be used when processing room impulse
+    responses. Alternatively see ``pyfar.dsp.find_impulse_response_delay``.
+
+    Parameters
+    ----------
+    impulse_response : pyfar.Signal
+        The impulse response
+    threshold : double, optional
+        Threshold according to ISO 3382 in dB, by default 20 dB.
+
+    Returns
+    -------
+    start_sample : numpy.ndarray, int
+        Sample at which the impulse response starts
+
+    Note
+    ----
```

Comment:
To my knowledge, we usually use either ``Notes`` (no blue box around it) or ".. note::" (blue box around it) in the docs. It the case of ``.. note::``, it is also comes before the parameters.
I'd slightly prefer to put it as a ``.. note::``.